### PR TITLE
feat(setup-elementor-env): init

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: actions/setup-wp-env/action.yml [hunk 1]
```diff
@@ -50,28 +50,17 @@ runs:
...
               INPUT_CONFIG: ${{ inputs.config }}
-              INPUT_WP_DEBUG: ${{ inputs.wp-debug }}
           run: |
               npx zx@8 $ACTION_PATH/dist/create-wp-env-config.js \
               --wp="$INPUT_WP" \
               --php="$INPUT_PHP" \
               --plugins="$INPUT_PLUGINS" \
               --themes="$INPUT_THEMES" \
               --active-theme="$INPUT_ACTIVE_THEME" \
-              --mappings="$INPUT_ACTIVE_THEME" \
```
Comment (StyleShit) [inline actions/setup-wp-env/action.yml:61]: 🫣

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
 	"scripts": {
 		"build": "tsup --config ./tsup.config.ts",
 		"dev": "npm run build -- --watch",
-		"lint": "concurrently -n eslint,tsc,prettier -c magenta,green,red \"eslint .\" \"tsc\" \"prettier . --check\""
+		"lint": "concurrently -n eslint,tsc,prettier -c magenta,green,red \"eslint .\" \"tsc\" \"prettier . --check\"",
```
Comment (StyleShit) [inline package.json:8]:
```suggestion
		"lint": "concurrently -n eslint,tsc,prettier -c magenta,blue,red \"eslint .\" \"tsc\" \"prettier . --check\"",
```

File: scripts/local-action.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+npx @github/local-action run "./actions/$1" index.ts "./actions/$1/.env"
```
Comment (StyleShit) [inline scripts/local-action.sh:1]: does it need to be here instead of inlined in the package json?

File: tsup.config.ts [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
 
-export default [defineActionConfig('setup-wp-env')];
-
-function defineActionConfig(action: string) {
-	return defineConfig({
-		entry: [`actions/${action}/*.ts`],
-		outDir: `actions/${action}/dist`,
+export default [
+	defineConfig({
+		entry: [`actions/setup-wp-env/*.ts`],
```
Comment (StyleShit) [inline tsup.config.ts:6]:
```suggestion
		entry: ['actions/setup-wp-env/*.ts'],
```

also others

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: .github/workflows/test-actions.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ concurrency:
 
 jobs:
     test-wp-env:
-        name: Setup WP Env
+        name: Setup WP and Elementor env
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply .github/workflows/test-actions.yml:12]: Done

File: actions/setup-elementor-env/action.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+name: Setup Elementor Environment
+description: Extends the setup-wp-env action to include Elementor specific configurations like experiments and templates.
+
+inputs:
+    env:
+        description: wp-env have 2 environments `development` and `testing`
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/action.yml]: Done

File: actions/setup-elementor-env/action.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+name: Setup Elementor Environment
+description: Extends the setup-wp-env action to include Elementor specific configurations like experiments and templates.
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/action.yml]: Done

File: actions/setup-elementor-env/action.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+inputs:
+    env:
+        description: wp-env have 2 environments `development` and `testing`
+        required: false
+        default: 'testing'
+    templates:
+        required: false
+        description: Paths to the Elementor templates directory, separated by new lines (e.g. `elementor-templates \n elementor-templates-2`)
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/action.yml]: Done

File: actions/setup-elementor-env/action.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    experiments:
+        required: false
+        description: A list of Elementor experiments to enable, separated by new lines (e.g. `atomic_widgets \n some_experiment`)
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/action.yml]: Done

File: actions/setup-elementor-env/main.ts [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+import { z } from 'zod';
+
+export async function run() {
+	try {
+		const { container, experiments, templates } = await core.group(
+			'Parsing inputs',
+			parseInputs,
+		);
+
+		await core.group('Validating setup-wp-env being used', async () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts]: Done

File: actions/setup-elementor-env/main.ts [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+			await runOnContainer({
+				container,
+				command: ['wp', 'core', 'version'],
+				error: '`setup-wp-env` is not being used - please use it in your workflow',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts]: Done

File: actions/setup-elementor-env/main.ts [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+			});
+		});
+
+		await core.group('Validating elementor being activated', async () => {
+			await runOnContainer({
+				container,
+				command: ['wp', 'plugin', 'is-active', 'elementor'],
+				error: 'Elementor is not installed - please define it in setup-wp-env',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts]: Done

File: actions/setup-elementor-env/main.ts [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+					container,
+					command: [
+						'wp',
+						'--user=admin',
+						'elementor',
+						'experiments',
+						'deactivate',
+						experiments.off.join(','),
+					],
+					error: 'Failed to deactivate experiments',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts]: Done

File: actions/setup-elementor-env/main.ts [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+						container,
+						command: [
+							'wp',
+							'--user=admin',
+							'elementor',
+							'library',
+							'import-dir',
+							template,
+						],
+						error: 'Failed to import templates',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts]: DOne

File: actions/setup-elementor-env/main.ts [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+		core.setFailed(error);
+	}
+}
+
+async function parseInputs() {
+	try {
+		const parsed = z
+			.object({
+				env: z.union([z.literal('development'), z.literal('testing')]),
+				templates: z.array(z.string().regex(/^[a-z0-9-_./]+$/)),
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts:106]: No, eslint yell at me

File: actions/setup-elementor-env/main.ts [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+			})
+			.parse({
+				env: core.getInput('env', { trimWhitespace: true }),
+				templates: core.getMultilineInput('templates', {
+					trimWhitespace: true,
+				}),
+				experiments: core
+					.getMultilineInput('experiments', { trimWhitespace: true })
+					.map((experiment) => experiment.split(':'))
+					.map(([key, value]) => [key, value?.toLowerCase()]),
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply actions/setup-elementor-env/main.ts]: Done

File: package.json (hunk 2 above)
Comment (Nevoss) [reply package.json]: Done

Comment (Nevoss) [reply package.json]: Done

File: scripts/local-action.sh (hunk 3 above)
Comment (Nevoss) [reply scripts/local-action.sh:1]: Not sure why, it does not work

File: tsup.config.ts (hunk 4 above)
Comment (Nevoss) [reply tsup.config.ts]: Done

File: actions/setup-elementor-env/main.ts [hunk 17]
```diff
@@ -54,7 +54,7 @@ export async function run() {
 						'deactivate',
 						experiments.off.join(','),
 					],
-					error: 'Failed to deactivate experiments',
+					error: `Failed to deactivate experiments: : ${experiments.on.join(', ')}`,
```
Comment (StyleShit) [inline actions/setup-elementor-env/main.ts:57]:
```suggestion
					error: `Failed to deactivate experiments: ${experiments.on.join(', ')}`,
```

**Review by StyleShit — APPROVED** [review]

**Pull request merged**